src/Cardano/Fee.hs

```diff
@@ -241,9 +251,12 @@ rebalanceChangeOutputs :: FeeOptions -> Fee -> [Coin] -> [Coin]
 rebalanceChangeOutputs opt totalFee chgs =
     case removeDust (Coin 0) chgs of
         [] -> []
-        xs -> removeDust (dustThreshold opt)
+        x : xs ->
```

Comment:
Here we can take advantage of the fact that we already have a non-empty list.

(We have already excluded the possibility of an empty list in the pattern match above this one.)